language_formatters_pre_commit_hooks/pretty_format_java.py

```diff
@@ -84,8 +84,29 @@ def pretty_format_java(argv: typing.Optional[typing.List[str]] = None) -> int:
         args.google_java_formatter_version,
     )
 
+    java_home_cmd = [
+        "/usr/libexec/java_home",
+        "-v",
+        "17"
+    ]
```

Comment:
Not quite familiar with the behavior here - what happens if there isn't a Java 17 VM installed?

Reply:
`-v` filters for the required version (e.g. 17). If that version is not installed, it defaults to the nearest installed version. 

If the user doesn't have 11 or higher installed, `pre-commit` will fail, similar to how we were failing with the old java formatter hook implementation: 

https://github.com/allocadia/Allocadia/blob/631e13417e6d9447142b1c87b3d3d4aa6257bf54/tools/style/bin/google-java-format.sh#L8